File: Source/C#/DevnologyFitnesseDojo/DevnologyFitnesseDojo/Fitnesse/Amasun/Book/CheckBookInventory.cs
using DevnologyFitnesseDojo.Domain;
using DevnologyFitnesseDojo.Domain.Books;

namespace DevnologyFitnesseDojo.Fitnesse.Amasun.Book
{
    public class CheckBookInventory
    {
        private string author;
        private string title;

        public int Amount()
        {
            var book = new Domain.Books.Book { Author = author, Title = title };

            return Inventory.CountBooks(book);
        }

        public void SetTitle(string title)
        {
            this.title = title;
        }

        public void SetAuthor(string author)
        {
            this.author = author;
        }
    }
}

File: Source/Java/nl/devnology/fitnesse/amasun/book/CheckBookInventory.java
package nl.devnology.fitnesse.amasun.book;

import nl.devnology.domain.book.Book;
import nl.devnology.domain.book.Inventory;

/**
 * @author Erik Pragt
 */
public class CheckBookInventory {
    private String author;
    private String title;
    private String isbn;
    private Integer price;


    public Integer amount() {
        Book book = new Book(author, title, isbn, price);

        return Inventory.get().countBooks(book);
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public void setAuthor(String author) {
        this.author = author;
    }

    public void setIsbn(String isbn) {
        this.isbn = isbn;
    }

    public void setPrice(Integer price) {
        this.price = price;
    }
}
